### PR TITLE
Mathml conversion using LutaML-Model

### LINKED ISSUE #302: MathML parsing and transformation using lutaml-model
Update **MathML** parsing and transformation structure from using **Ox** (for parsing input) and **Parslet** (for transforming parsed input into a formula) to **Lutaml-Model**.

### PR DESCRIPTION
**Pull request by suleman-uzair**
This PR updates the parsing structure for **MathML** from using **Ox**(for parsing) and **Parslet** (for transformation).

close #302

### CODE REVIEW

File: lib/plurimath/math/formula/mrow.rb [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+module Plurimath
+  module Math
+    class Formula
+      class Mrow < Formula
```
Comment (ronaldtse) [inline lib/plurimath/math/formula/mrow.rb:6]: Is this `mrow` (and `mstyle`) a concept corresponding to MathML or is this to represent MathML? i.e. is this a generic concept? If it's a generic concept, should we just call it `row` and `style`?

File: lib/plurimath/math/function/menclose.rb [hunk 2]
```diff
@@ -1,11 +1,14 @@
 # frozen_string_literal: true
 
 require_relative "binary_function"
+require_relative "../../mathml/utility"
 
 module Plurimath
   module Math
     module Function
       class Menclose < BinaryFunction
```
Comment (ronaldtse) [inline lib/plurimath/math/function/menclose.rb:9]: Similar to my comment above whether this is generic or MathML specific?

File: lib/plurimath/math/formula/mrow.rb (hunk 1 above)
Comment (suleman-uzair) [reply lib/plurimath/math/formula/mrow.rb:6]: @ronaldtse, no, they are not generic and exist only for **MathML** parsing using **LutaML-Model**.

File: lib/plurimath/math/function/menclose.rb (hunk 2 above)
Comment (suleman-uzair) [reply lib/plurimath/math/function/menclose.rb:9]: This class itself is not generic but other conversions output are supported.

**Review by HassanAkbar — APPROVED** [review]

**Comment by ronaldtse** [comment]
@suleman-uzair I have a question. For files like:
* lib/plurimath/math/function/fenced.rb

Is it possible to directly utilize lutaml-model to handle them?

Actually now we are talking about a `Plurimath::Math` model transformation into the `Mml` model which goes into MathML XML.

i.e.
```ruby
module Plurimath
  module Math
    module Function
      class TernaryFunction < Lutaml::Model::Serializable
        attribute :parameter_one, Math::Function::Core
        attribute :parameter_two, Math::Function::Core
        attribute :parameter_three, Math::Function::Core
        attribute :hide_function_name, :boolean

        def initialize(*args)
          super
          Utility.validate_left_right(variables.map { |var| get(var) })
        end
# ...
end

module Plurimath
  module Math
    module Function
      class Fenced < TernaryFunction
        include Mathml::Utility

        attribute :separators, Math::Symbol, validate: -> { |value| validate_symbols(value) }
        attribute :content, Math::Function::Core
        attribute :open, Math::Symbol, validate: -> { |value| validate_symbols(value) }
        attribute :close, Math::Symbol, validate: -> { |value| validate_symbols(value) }
        # ...
end
```

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
@suleman-uzair since this is done let's merge it first. Thank you for the good work!

File: Gemfile [hunk 3]
```diff
@@ -13,3 +13,4 @@ gem 'oga'
 gem 'ox'
 gem "unitsml"
 gem "debug"
+gem "mml", git: "https://github.com/plurimath/mml", branch: "features/mathml_parsing"
```
Comment (ronaldtse) [inline Gemfile:16]: We need to remove the git address, I'm sure you will...

Comment (suleman-uzair) [reply Gemfile:16]: > We need to remove the git address

@ronaldtse, Yes, I’ll do that but we’ll need to release the `mml` gem’s version before that.
Can you review/approve the plurimath/mml#2 first, so I can merge and release the version first?

**Pull request merged**